prow/report/report.go

```diff
@@ -45,20 +45,39 @@ type GithubClient interface {
 	EditComment(org, repo string, ID int, comment string) error
 }
 
+// prowjobStateToGithubStatus maps prowjob status to github states.
+// Github states can be one of error, failure, pending, or success.
+// https://developer.github.com/v3/repos/statuses/#create-a-status
+func prowjobStateToGithubStatus(pjState v1.ProwJobState) (string, error) {
+	switch pjState {
+	case v1.TriggeredState:
+		return github.StatusPending, nil
+	case v1.PendingState:
+		return github.StatusPending, nil
+	case v1.SuccessState:
+		return github.StatusSuccess, nil
+	case v1.ErrorState:
+		return github.StatusError, nil
+	case v1.FailureState:
+		return github.StatusFailure, nil
+	case v1.AbortedState:
+		return github.StatusFailure, nil
```

Comment:
It is unclear to me if this should be marked as a failure or an error, but I don't think it matters too much.